client/dashboard/me/profile-personal-details/update-email/email-verification-banner.tsx

```diff
@@ -0,0 +1,129 @@
+import { userSettingsQuery, resendEmailVerificationMutation } from '@automattic/api-queries';
+import { useMutation, useSuspenseQuery } from '@tanstack/react-query';
+import { Button } from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { createInterpolateElement } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useState, useEffect } from 'react';
+import Notice from '../../../components/notice';
+
+export default function EmailVerificationBanner() {
+	const { data: userData } = useSuspenseQuery( userSettingsQuery() );
+	const { createErrorNotice } = useDispatch( noticesStore );
+	const [ showResendButton, setShowResendButton ] = useState( true );
+	const [ showSuccessNotice, setShowSuccessNotice ] = useState( false );
+
+	const isEmailPending = userData.user_email_change_pending;
+	const pendingEmail = userData.new_user_email;
+
+	// Handle verification success/error
+	useEffect( () => {
+		if ( typeof window === 'undefined' ) {
+			return;
+		}
+
+		const params = new URLSearchParams( window.location.search );
+		const newEmailResult = params.get( 'new_email_result' );
+
+		if ( newEmailResult === '0' ) {
+			createErrorNotice(
+				__( 'The email verification link is invalid or has expired. Please request a new one.' ),
+				{ type: 'snackbar' }
+			);
+
+			params.delete( 'new_email_result' );
+			const newUrl =
+				window.location.pathname + ( params.toString() ? '?' + params.toString() : '' );
+			window.history.replaceState( {}, '', newUrl );
+		} else if ( newEmailResult === '1' ) {
+			setShowSuccessNotice( true );
+
+			params.delete( 'new_email_result' );
+			const newUrl =
+				window.location.pathname + ( params.toString() ? '?' + params.toString() : '' );
+			window.history.replaceState( {}, '', newUrl );
+		}
+	}, [ createErrorNotice ] );
+
+	// Resend email
+	const { mutate: resendEmail, isPending: isResendPending } = useMutation( {
+		...resendEmailVerificationMutation(),
+		onSuccess: () => {
+			setShowResendButton( false );
+		},
+		onError: () => {
+			setShowResendButton( true );
+		},
+		meta: {
+			snackbar: {
+				success: pendingEmail
+					? sprintf(
+							/* translators: %s is the user's new email address they're trying to change to */
+							__( 'We sent an email to %s. Please check your inbox to verify your email.' ),
+							pendingEmail
+					  )
+					: __( 'Verification email sent.' ),
+				error: __( 'Failed to resend verification email.' ),
+			},
+		},
+	} );
+
+	const handleResendEmail = () => {
+		if ( ! pendingEmail ) {
+			return;
+		}
+		resendEmail( pendingEmail );
+	};
+
+	if ( showSuccessNotice ) {
+		return (
+			<Notice
+				variant="success"
+				title={ __( 'Email address updated' ) }
+				onClose={ () => setShowSuccessNotice( false ) }
+				actions={
+					<Button variant="link" href="/v2/domains/">
+						{ __( 'Update domain contacts' ) }
+					</Button>
+				}
```

Comment:
```suggestion
				actions={ <Link to="/domains/">{ __( 'Update domain contacts' ) }</Link> }
```

I noticed that clicking the domains link will trigger a full page refresh. Using `import { Link } from '@tanstack/react-router';` should fix that

Reply:
Ahh and using `Button` here isn't accessible either 🙈 Updated to use `<Link>`, thank you!! [8bcc572](https://github.com/Automattic/wp-calypso/pull/106103/commits/8bcc572e5f88b4609bb7f7b2ca455352ccc9ef8b)